global_run.py

```diff
@@ -0,0 +1,82 @@
+import random
+from copy import deepcopy
+import pickle as pkl
+from tqdm import tqdm
+import os
+import time
+
+
+def subsample(samples, n=1000):
+    selected_idxes = list(range(len(samples)))
+    random.shuffle(selected_idxes)
+    selected_idxes = selected_idxes[:n]
+    return [samples[i] for i in sorted(selected_idxes)]
+
+# flip the problem of describing corpus A to describing corpus B
+def flip_problem(problem):
+    problem = deepcopy(problem)
+    problem['A_desc'], problem['B_desc'] = problem['B_desc'], problem['A_desc']
+    problem['split'] = {
+        k: {
+            'A_samples': v['B_samples'],
+            'B_samples': v['A_samples']
+        } for k, v in problem['split'].items()
+    }
+    return problem
+
+
+if __name__ == '__main__':
+    from D5 import D5
+    from validator import DummyValidator, Validator
+    from lm_proposer import GPT3_Proposer
+    from get_representative import return_extreme_values
+
+    problems = pkl.load(open('OpenD5.pkl', 'rb'))
+
+    # the default validator has 11B parameters
+    validator = Validator()
+    if not os.path.exists('discoveries'):
+        os.mkdir('discoveries')
+
+    # randomly shuffle the problems to run our system on
+    problem_idxes = list(range(len(problems)))
+    random.shuffle(problem_idxes)
+    pbar = tqdm(problem_idxes)
+
+
+    for problem_id in pbar:
+        pbar.set_description(f'problem {problem_id}')
+        problem_orig = problems[problem_id]
+
+        def get_h2h_dicts(problem, save_path):
+            current_time = time.time()
+            if os.path.exists(save_path):
+                loaded_object = pkl.load(open(save_path, 'rb'))
+                if type(loaded_object) == dict:
+                    return loaded_object
+                elif current_time - loaded_object < 60 * 60 * 8:
+                    return None
+            pkl.dump(current_time, open(save_path, 'wb'))
+
+            extreme_vals = return_extreme_values(problem['split']['research']['A_samples'], problem['split']['research']['B_samples'])
+            problem['split']['research']['A_samples'] = subsample(extreme_vals['sorted_A_samples'])
```

Comment:
On a more general note a `problem` seems like an important object in your code, and they make up the OpenD5 which is one of your main research outputs. 

It should probably be its own class. You could make use of pythons `dataclasses`. If you want to combine this with the above json comment above using something like https://pypi.org/project/dataclasses-json/.